File: haskell-server/test/EagerWords/Server/Domain/BoardSpec.hs
--
-- Copyright 2017-2020 Azad Bolour
-- Licensed under GNU Affero General Public License v3.0 -
--   https://github.com/azadbolour/eagerwords/blob/master/LICENSE.md
--

module EagerWords.Server.Domain.BoardSpec where

import Test.Hspec
import Data.Either
import Bolour.Util.SpecUtil (satisfiesRight, satisfiesLeft)
import qualified Bolour.Plane.Domain.Grid as Grid
import Bolour.Plane.Domain.Point (Point, Point(Point))
import EagerWords.Server.Domain.Board (validatePoint)
import qualified EagerWords.Server.Domain.Board as Board

spec :: Spec
spec = do
  describe "Make Board" $ do
    let b = Board.mkEmptyBoard 10
    it "has edge points" $ do
       satisfiesRight $ validatePoint b $ Point 0 0
       satisfiesRight $ validatePoint b $ Point 9 0
       satisfiesRight $ validatePoint b $ Point 0 9
       satisfiesRight $ validatePoint b $ Point 9 4
       return ()
    it "does not have external points" $ do
       satisfiesLeft $ validatePoint b $ Point (-1) 0
       satisfiesLeft $ validatePoint b $ Point 10 0
       satisfiesLeft $ validatePoint b $ Point 0 10
       satisfiesLeft $ validatePoint b $ Point 9 11
       return ()

File: scala-server/src/test/scala/com/bolour/eagerwords/server/domain/BoardSpec.scala
/*
 * Copyright 2017-2020 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *    https://github.com/azadbolour/eagerwords.com/blob/master/LICENSE.md
 */
package com.bolour.eagerwords.server.domain

import org.scalatest.{FlatSpec, Matchers}
import org.slf4j.LoggerFactory

import com.bolour.grid.domain.Point
import com.bolour.eagerwords.common.domain.{Piece, PiecePoint}

class BoardSpec extends FlatSpec with Matchers {
  val logger = LoggerFactory.getLogger(this.getClass)

  "board" should "start empty" in {
    val board1 = Board(15)
    board1.piecePoints.size shouldEqual 0

    val noPiecePoints: List[PiecePoint] = Nil
    val board2: Board = Board(2, noPiecePoints)
    board2.piecePoints.size shouldBe 0

    val piecePoints = List(PiecePoint(Piece('A', "idA"), Point(0, 0)))
    val board3 = Board(1, piecePoints)
    logger.info(s"${board3}")
    board3.piecePoints.size shouldEqual 1
  }
}
